libs/datamodel/core/src/configuration/configuration.rs

```diff
@@ -1,6 +1,21 @@
 use super::{Datasource, Generator};
+use crate::error::{DatamodelError, ErrorCollection};
 
 pub struct Configuration {
     pub generators: Vec<Generator>,
     pub datasources: Vec<Datasource>,
 }
+
+impl Configuration {
+    pub fn validate_that_one_datasource_is_provided(self) -> Result<Self, ErrorCollection> {
+        if self.datasources.is_empty() {
+            Err(DatamodelError::new_validation_error(
+                "You defined no datasource. You must define exactly one datasource.",
+                crate::ast::Span::new(0, 5),
```

Comment:
I am curious about what it does with an empty datamodel (it shouldn't happen, but just in case)

Reply:
Hmm good point. The pretty printing might crash. We will find out at some point :-)